### PR TITLE
rustbuild: Skip generating docs if the config disables them.

### PR DESCRIPTION
**Pull request by timonvo**
r? @alexcrichton

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks for the PR! I think the [strategy here](https://github.com/rust-lang/rust/pull/32599), however, may be the more manageable approach where we just don't generate these targets in those situations (but they can still be explicitly run if desired).

Want to update the PR here to that strategy? May be able to land the two in parallel to see which hits master first.

**Comment by timonvo** [comment]
Yeah, that does look a better approach. I've updated the branch to use that approach. I see that the other PR may be close to ready though, so if that's the case we can just close the PR and let the fix get submitted through the other one.

**Comment by timonvo** [comment]
Also note that it wasn't really clear to me if or how the build.config.compiler_docs variable should be used now. I'm not sure what steps constitute "compiler docs" vs non-compiler docs.

File: src/bootstrap/build/step.rs [hunk 1]
```diff
@@ -148,7 +148,9 @@ fn top_level(build: &Build) -> Vec<Step> {
             src: Source::Llvm { _dummy: () },
             target: &build.config.build,
         };
-        targets.push(t.doc(stage));
+        if build.config.docs {
+          targets.push(t.doc(stage));
```
Comment (alexcrichton) [inline src/bootstrap/build/step.rs]: Could you use 4 space tabs here as well?

Comment (timonvo) [inline src/bootstrap/build/step.rs]: Done in latest commit. I'm assuming the commits will be squashed before being merged, right?

Comment (alexcrichton) [inline src/bootstrap/build/step.rs]: Indeed!

**Comment by alexcrichton** [comment]
Could you squash the commits down into one as well? Also feel free to ping a PR after it's updated because unfortunately github doesn't send out notifications for that :(

**Comment by timonvo** [comment]
Squashed into a single commit now.

**Comment by alexcrichton** [comment]
@bors: r+ 2805e83dcb3762994b8802fa766c88ca93c5b839

Thanks!

**Pull request merged**